src/test/java/io/confluent/idesidecar/restapi/util/AbstractSidecarIT.java

```diff
@@ -0,0 +1,241 @@
+
+import static io.confluent.idesidecar.restapi.kafkarest.SchemaManager.SCHEMA_PROVIDERS;
+
+import io.confluent.idesidecar.restapi.messageviewer.RecordDeserializer;
+import io.confluent.idesidecar.restapi.messageviewer.SimpleConsumer;
+import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient;
+import io.quarkus.logging.Log;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Predicate;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.junit.jupiter.api.AfterEach;
+
+/**
+ * Abstract base class for integration tests that require a {@link TestEnvironment} and use
+ * a sidecar client to interact with the Sidecar REST API.
+ *
+ * <h2>Sharing the {@link TestEnvironment}</h2>
+ * <p>All subclasses of this class will share the same {@link LocalTestEnvironment} instance,
+ * which is started before any tests are run and stopped after all tests have run.
+ * Starting the test environment's containers takes 5-10 seconds, so doing it once
+ * <i>for all integration tests</i> (that extend this class) helps the tests run faster.
+ *
+ * <h2>Sharing or not sharing connections</h2>
+ * <p>Subclasses are expected to set up the sidecar connection(s) that they want their test
+ * methods to use. Creating a connection may take several seconds, depending upon the effort
+ * require to validate the connection and determine the status. This class provides a flexible
+ * way for subclasses to control how connections are created and reused.
+ *
+ * <p>Most subclasses will use the same connection for all tests, and can use
+ * have the test class' {@code @BeforeEach} method call either
+ * {@link #setupLocalConnection(Class)} or {@link #setupDirectConnection(Class)}, which will
+ * reuse the same connection for all tests in the class.
+ * This minimizes the setup time for each test, as only the first test will need to create the
+ * (shared) connection.
+ *
+ * <p>Other tests may want to use a different connection for each test, and can use
+ * have the test class' {@code @BeforeEach} method call either
+ * {@link #setupLocalConnection(String)} or {@link #setupDirectConnection(String)} to create
+ * connections for a specified "test scope". This allows tests to use _different_ connections,
+ * but reusing connections depends on which tests reuse the same scopes.
+ *
+ * <p>And yet other integration tests may want each test to set up the connections it needs.
+ * In this case, each test should call the appropriate {@code setup*Connection} method and, when
+ * finished, call {@link #deleteAllConnections(Predicate)} to clean up the test's connections.
+ * This approach is the most flexible, but also the slowest, as each test will need to create
+ * the connections it needs.
+ */
+public abstract class AbstractSidecarIT extends SidecarClient {
```

Comment:
The `ConfluentLocalTestBed` class was broken up into this class, the `SidecarClient` class that provides easy-to-use functions that tests can use to submit requests to the sidecar APIs, and the `LocalTestEnvironment` for managing the connections.